server/lib/transactions.ts

```diff
@@ -213,6 +215,84 @@ export async function createTransactionsFromPaidExpense(
   return models.Transaction.createDoubleEntry(transaction);
 }
 
+export async function createTransactionsForManuallyPaidExpense(
+  host,
+  expense,
+  paymentProcessorFeeInHostCurrency,
+  totalAmountPaidInHostCurrency,
+  /** Will be stored in transaction.data */
+  transactionData: Record<string, unknown> = {},
+) {
+  assert(paymentProcessorFeeInHostCurrency >= 0, 'Payment processor fee must be positive');
+  assert(totalAmountPaidInHostCurrency >= 0, 'Total amount paid must be positive');
+
+  const isCoveredByPayee = expense.feesPayer === 'PAYEE';
+  const amount = toNegative(totalAmountPaidInHostCurrency - paymentProcessorFeeInHostCurrency);
+  const netAmountInCollectiveCurrency = toNegative(totalAmountPaidInHostCurrency);
+  const amounts = {
+    amount,
+    amountInHostCurrency: amount,
+    paymentProcessorFeeInHostCurrency: toNegative(paymentProcessorFeeInHostCurrency),
+    netAmountInCollectiveCurrency,
+    hostCurrencyFxRate: 1,
+  };
+
+  if (host.currency !== expense.collective.currency) {
+    assert(
+      expense.currency === expense.collective.currency,
+      'Expense currency must be the same as collective currency',
+    );
+    amounts.hostCurrencyFxRate = isCoveredByPayee
+      ? // If the payee is covering the fees, the expense amount should be proportional to the total amount paid in host currency
+        expense.amount / totalAmountPaidInHostCurrency
+      : // If the host is covering the fees, the expense amount be proportional to total amount paid minus the fees
```

Comment:
I realize I made a confusing comment when we discussed this on Slack:
- When paying an expense, unless `feesPayer=PAYEE`, the fee is paid by the collective. 
   _Example: A collective with $200 balance, expense is $80, fees are $20 => The collective balance will be $100 after paying._
- When refunding an expense (aka. "Mark as unpaid") the host is indeed covering the fees (via a `PAYMENT_PROCESSOR_COVER` transaction). The intent is to make sure the full amount is refunded to the collective.
   _Example: Host has a $1000 balance. Collective with $200 balance, expense is $80, fees are $20 => The collective balance will be $100 after paying. The collective balance will be back to $200 if the expense is refunded, but the host's will be $1000 - $20 = $980_

```suggestion
      : // If the collective is covering the fees, the expense amount should be proportional to total amount paid minus the fees
```

Reply:
You're correct. But on top of that, there's just a misconception here.
This is not about the logic of who's paying for the fees, I was just a bit confused about what value to use when calculating the FX rate.
Turns out, the correct here is to always consider the `expense.amount / amount`.